src/main/kotlin/gg/flyte/christmas/minigame/games/Spleef.kt

```diff
@@ -203,54 +352,324 @@ class Spleef : EventMiniGame(GameConfig.SPLEEF) {
 
         listeners += event<PlayerToggleFlightEvent> {
-            if (player.gameMode != GameMode.SURVIVAL) return@event // elimed players alr in ADVENTURE
+            if (player.gameMode != GameMode.SURVIVAL) return@event
             isCancelled = true
 
-            if (doubleJumps[player]!! > 0) {
-                player.isFlying = false
-                player.velocity = player.location.direction.multiply(0.5).add(Vector(0.0, 1.25, 0.0))
-                player.playSound(Sound.ITEM_TOTEM_USE)
-                doubleJumps[player] = doubleJumps[player]!! - 1
-            } else player.allowFlight = false
+            if (unlimitedJumps) {
+                doubleJump(player)
+            } else if (doubleJumps[player.uniqueId]!! > 0) {
+                doubleJump(player)
+
+                doubleJumps[player.uniqueId] = doubleJumps[player.uniqueId]!! - 1
+            } else {
+                player.allowFlight = false
+            }
+        }
+
+        listeners += event<ProjectileLaunchEvent> {
+            if (entity !is Snowball) return@event
+
+            if (powerfulSnowballs) {
+                entity.velocity = entity.velocity.multiply(2)
+            }
         }
 
         listeners += event<ProjectileHitEvent> {
+            if (hitEntity != null && entity.shooter !is Player) {
+                isCancelled = true // make snowballs fly through entities to increase chances of spleefing
+            }
+
             if (hitBlock == null) return@event
             if (entity !is Snowball) return@event
 
-            floorLevelBlocks.any { it.block == hitBlock }.let {
-                when (hitBlock!!.type) {
-                    Material.SNOW_BLOCK -> {
-                        hitBlock!!.type = Material.SNOW
-                        if (hitBlock!!.blockData !is Snow) return@event
-                        hitBlock!!.blockData = (hitBlock!!.blockData as Snow).apply { layers = 5 }
-                    }
+            if (floorLevelBlocks.any { it.block == hitBlock }) {
+                if (powerfulSnowballs) {
+                    hitBlock!!.type = Material.AIR
+                } else {
+                    wearDownSnowBlock(hitBlock!!)
+                }
+            }
 
-                    Material.SNOW -> {
-                        val blockData = hitBlock!!.blockData as Snow
-                        if (blockData.layers == 2) {
-                            hitBlock!!.type = Material.AIR
-                        } else {
-                            blockData.layers = blockData.layers - 3
-                            hitBlock!!.blockData = blockData
-                        }
-                    }
+            // implies snowball source is snowball rain
+            if (entity.shooter !is Player && entity.shooter !is Snowman) {
+                isCancelled = true // make snowball rain destroy multiple blocks
+            }
+        }
 
-                    else -> return@event
+        listeners += event<PlayerDropItemEvent> {
```

Comment:
This is already handled in housekeeping.

Reply:
Ah yeah I was confused when /event mock-donation removed an item from my inventory, I thought I accidentally dropped it. Will remove this.